x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/aggregate/ValuesTests.java

```diff
@@ -0,0 +1,112 @@
+import org.elasticsearch.xpack.esql.expression.function.TestCaseSupplier;
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.nullValue;
+
+public class ValuesTests extends AbstractAggregationTestCase {
+    public ValuesTests(@Name("TestCase") Supplier<TestCaseSupplier.TestCase> testCaseSupplier) {
+        this.testCase = testCaseSupplier.get();
+    }
+
+    @ParametersFactory
+    public static Iterable<Object[]> parameters() {
+        var suppliers = new ArrayList<TestCaseSupplier>();
+
+        Stream.of(
+            MultiRowTestCaseSupplier.intCases(1, 1000, Integer.MIN_VALUE, Integer.MAX_VALUE, true),
+            MultiRowTestCaseSupplier.longCases(1, 1000, Long.MIN_VALUE, Long.MAX_VALUE, true),
+            MultiRowTestCaseSupplier.doubleCases(1, 1000, -Double.MAX_VALUE, Double.MAX_VALUE, true),
+            MultiRowTestCaseSupplier.dateCases(1, 1000),
+            MultiRowTestCaseSupplier.booleanCases(1, 1000),
+            MultiRowTestCaseSupplier.ipCases(1, 1000),
+            MultiRowTestCaseSupplier.versionCases(1, 1000),
+            // Lower values for strings, as they take more space and may trigger the circuit breaker
+            MultiRowTestCaseSupplier.stringCases(1, 100, DataType.KEYWORD),
+            MultiRowTestCaseSupplier.stringCases(1, 100, DataType.TEXT)
+        ).flatMap(List::stream).map(ValuesTests::makeSupplier).collect(Collectors.toCollection(() -> suppliers));
+
+        // TODO: Keyword, text
+
+        return parameterSuppliersFromTypedDataWithDefaultChecks(suppliers);
+    }
+
+    @Override
+    protected Expression build(Source source, List<Expression> args) {
+        return new Values(source, args.get(0));
+    }
+
+    @SuppressWarnings("unchecked")
+    private static TestCaseSupplier makeSupplier(TestCaseSupplier.TypedDataSupplier fieldSupplier) {
+        return new TestCaseSupplier(fieldSupplier.name(), List.of(fieldSupplier.type()), () -> {
+            var fieldTypedData = fieldSupplier.get();
+
+            var expected = fieldTypedData.multiRowData()
+                .stream()
+                .map(v -> (Comparable<? super Comparable<?>>) v)
+                .collect(Collectors.toSet());
+
+            return new TestCaseSupplier.TestCase(
+                List.of(fieldTypedData),
+                "Values[field=Attribute[channel=0]]",
+                fieldSupplier.type(),
+                expected.isEmpty() ? nullValue() : valuesInAnyOrder(expected)
+            );
+        });
+    }
+
+    private static <T, C extends T> Matcher<Object> valuesInAnyOrder(Collection<T> data) {
```

Comment:
This may be removed after https://github.com/elastic/elasticsearch/pull/109017/files#diff-c3559c9c9cfba4efabacf27d168f5a866670abbf5a67a783a76454a48869e185R1297